numba/typing/ctypes_utils.py

```diff
@@ -113,9 +113,10 @@ def make_function_type(cfnptr):
     cargs = [from_ctypes(a)
              for a in cfnptr.argtypes]
     cret = from_ctypes(cfnptr.restype)
-    # void* return type is implicitly converted to intp to match python
+    # void* return type is a int/long on 32 bit platforms and an int on 64 bit
+    # platforms, explicit conversion to a int64 should match.
     if cret == types.voidptr:
-        cret = types.intp
+        cret = types.int64
```

Comment:
`long` on 32-bit?  But the code is using `int64` always.  `ctypes.sizeof(ctypes.c_long)==4` on windows.

Reply:
yeah, this is wrong, `intp` is also not working as it causes `OverflowError, Python int too large to convert to C long` on a variety of platforms.